cleandart/react-dart
# Reduce DDC SyntheticEvent warning noise

## Motivation
As a workaround to https://github.com/dart-lang/sdk/issues/43939, our `SyntheticEvent` classes are non `@anonymous`.

A consequence of this is that, in the DDC, when event handlers are type-checked in certain situations, they emit console warnings every time.

These warnings can quickly become excessive when interacting with an app.
![Screen Shot 2020-12-16 at 4 44 15 PM](https://user-images.githubusercontent.com/1713967/102527454-590a1780-405a-11eb-9650-d0c9ead84a6c.png)

## Solution
- In dev builds of React, wrap console.warn and only emit errors for SyntheticEvent wrappers once
    - Also log an additional message indicating that the warning is expected, and linking to the relevant Dart issue
- Add tests

Boyscouting
- Update outdated name of js_interop_helpers_test test directory and outdated README

## Testing
- Verify tests pass in both DDC and dart2js
    - DDC (run `pbr test -- -P dartdevc`)
    - dart2js (run `pbr test -r -- -P dart2js`)
- Verify that the tests properly test this behavior
    - Revert compiled JS changes (commit e2fe5a4)
   - Verify that `console.warn` tests fail in DDC (`pbr test -- -P dartdevc`) for only the "dev" suites (prod files should not be affected)
- Pull this branch into an app where this issue occurs and verify that the error is only logged once
    -  e.g., run the w_context_menu web directory and hover/select items in the context menu
